tests/libvmi/factory.go

```diff
@@ -47,27 +46,49 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
 
 // NewCirros instantiates a new CirrOS based VMI configuration
 func NewCirros(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	diskImage := WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskCirros))
+
+	opts = append(opts, diskImage)
+	cirrosOpts := NewCirrosOptions(opts...)
+
+	return New(RandName(DefaultVmiName), cirrosOpts...)
+}
+
+// NewCirrosOptions instantiates default options needed for a CirrOS based VMI configuration
+func NewCirrosOptions(opts ...Option) []Option {
 	// Supplied with no user data, Cirros image takes 230s to allow login
 	withNonEmptyUserData := WithCloudInitNoCloudUserData("#!/bin/bash\necho hello\n", true)
 
+	// TODO: Check this issue with Cirros Memory requirements
+	// Cirros image need 256M to boot on ARM64,
+	// this issue is traced in https://github.com/kubevirt/kubevirt/issues/6363
 	cirrosOpts := []Option{
-		WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskCirros)),
-		withNonEmptyUserData,
 		WithResourceMemory("128Mi"),
 		WithTerminationGracePeriod(DefaultTestGracePeriod),
 	}
 	cirrosOpts = append(cirrosOpts, opts...)
-	return New(RandName(DefaultVmiName), cirrosOpts...)
+	// put userData  - cloudInit last, so it is after any user provided disks
+	cirrosOpts = append(cirrosOpts, withNonEmptyUserData)
+	return cirrosOpts
 }
 
 // NewAlpine instantiates a new Alpine based VMI configuration
 func NewAlpine(opts ...Option) *kvirtv1.VirtualMachineInstance {
+	diskImage := WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+
+	opts = append(opts, diskImage)
+	alpineOpts := NewAlpineOptions(opts...)
+
+	return New(RandName(DefaultVmiName), alpineOpts...)
+}
+
+// NewCirrosOptions instantiates default options needed for a CirrOS based VMI configuration
+func NewAlpineOptions(opts ...Option) []Option {
```

Comment:
The `opts` parameter is only passed through. Can you remove it, and leave it up to the caller to append any other paremeters?

When you rebase this PR on the latest `main`, there is a new function `NewAlpineWithTestTooling`, which could also use this function.

Reply:
Thanks, good idea.

Reply:
Well, actually I would like to keep it as a Decorator function. 
So the caller provides parameters, for example as `opts`, and then wraps/decorates it into Alpine flavor:
```
opts : = someUserProvidedOptionsAnddisks()
vmi := New(RandName(DefaultVmiName), NewAlpineOptions(opts...))

```

That way options provided by caller always take precedence (for example one can provide a different ContainerImage, first one provided wins). That way we can also do some checks/validations/add things that have to go last - as in Cirros.

Reply:
I have rebased and done some cleanup, WDYT?